pypeit/spectrographs/soar_goodman.py

```diff
@@ -0,0 +1,306 @@
+
+class SOARGoodmanRedSpectrograph(SOARGoodmanSpectrograph):
+    name = 'soar_goodman_red'
+    camera = 'red'
+    comment = 'Supported gratings: M1, M2 and 2x2 binning'
+    supported = True
+
+    def get_detector_par(self, hdu, det):
+        """
+        Return metadata for the selected detector.
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`_):
+                The open fits file with the raw image of interest.
+            det (:obj:`int`):
+                1-indexed detector number.
+
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+        header = hdu[0].header
+        # Binning
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning')  # Could this be detector dependent??
+
+        # Detector 1
+        detector_dict = dict(
+            binning         = binning,
+            det             = 1,
+            dataext         = 0,
+            specaxis        = 1,
+            specflip        = False,
+            spatflip        = False,
+            platescale      = 0.15,
+            darkcurr        = 0.00008,  # e-/s/pix
+            saturation      = 65535.,
+            nonlinear       = 1.0,
+            mincounts       = -1e10,
+            numamplifiers   = 1,
+            gain            = np.atleast_1d(header['GAIN']),
+            ronoise         = np.atleast_1d(header['RDNOISE']),
+            )
+        
+        # Only tested for 2x2
+        if binning == '2,2':
+            # parse TRIMSEC
+            col0 = int(header['TRIMSEC'][1:].split(':')[0])
+            dsec = f"[:,{col0*2}:]"  # rows, columns on the raw frame
+            detector_dict['datasec'] = np.atleast_1d(dsec)
+            # Overscan
+            osec = f"[:,1:{int(col0*2)-2}:]"
+            detector_dict['oscansec'] = np.atleast_1d(osec)
+        else:
+            msgs.error("Ask the developers to add your binning.  Or add it yourself.")
+        
+        # Return
+        return detector_container.DetectorContainer(**detector_dict)
+
+    @classmethod
+    def default_pypeit_par(cls):
+        """
+        Return the default parameters to use for this instrument.
+        
+        Returns:
+            :class:`~pypeit.par.pypeitpar.PypeItPar`: Parameters required by
+            all of ``PypeIt`` methods.
+        """
+        par = super().default_pypeit_par()
+
+        # Turn off bias and turn on overscan
+        turn_off_on = dict(use_biasimage=False, use_darkimage=False, use_overscan=True)
+        par.reset_all_processimages_par(**turn_off_on)
+
+        # Ignore PCA
+        par['calibrations']['slitedges']['sync_predict'] = 'nearest'
+
+        # Set pixel flat combination method
+        #par['calibrations']['pixelflatframe']['process']['combine'] = 'median'
+        # Change the wavelength calibration method
+        par['calibrations']['wavelengths']['method'] = 'holy-grail'
```

Comment:
Would it be worthwhile adding your own wavelength solutions to the reidentify database, too? Depends on the relative wavelength coverage you have...

Reply:
Good suggestion and I'll do this if we find Holy Grail 
doesn't hold for others.